lib/block-supports/typography.php

```diff
@@ -206,6 +206,83 @@ function gutenberg_typography_get_css_variable_inline_style( $attributes, $featu
 	return sprintf( '%s:var(--wp--preset--%s--%s);', $css_property, $css_property, $slug );
 }
 
+/**
+ * Returns a font-size value based on a given font-size preset. If typography.fluid is enabled it will calculate clamp values.
+ *
+ * @param array $preset Duotone preset value as seen in theme.json.
+ * @return string        Font-size value.
+ */
+function gutenberg_get_typography_font_size_value( $preset ) {
+	$typography_settings = gutenberg_get_global_settings( array( 'typography' ) );
+
+	// This is where we'll keep options I guess.
+	if ( ! isset( $typography_settings['fluid'] ) ) {
+		return $preset['size'];
+	}
+
+	// Up for discussion.
+	$default_unit                   = 'rem';
+	$default_minimum_viewport_width = '1600px';
+	$default_maximum_viewport_width = '650px';
+
+	// Matches rem or px values only.
+	$pattern = '/^(\d*\.?\d+)(rem|px)?$/';
+	// Could we also take these from layout? contentSize and wideSize?
+	$minimum_viewport_width      = isset( $typography_settings['minViewportWidth'] ) ? $typography_settings['minViewportWidth'] : $default_minimum_viewport_width;
+	$minimum_viewport_width_unit = $default_unit;
+	$maximum_viewport_width      = isset( $typography_settings['maxViewportWidth'] ) ? $typography_settings['maxViewportWidth'] : $default_maximum_viewport_width;
+	$maximum_viewport_width_unit = $default_unit;
+
+	// Minimum viewport size.
+	preg_match_all( $pattern, $minimum_viewport_width, $minimum_viewport_width_matches );
```

Comment:
Test, extract and set the value and unit variables... all ripe for abstraction 
Maybe a method that pulls out what we need / sets a fallback.